docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Drupal\foia_webform;
+
+use Drupal\file\Entity\File;
+use Drupal\node\NodeInterface;
+use Drupal\webform\WebformSubmissionInterface;
+use GuzzleHttp\ClientInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class FoiaSubmissionServiceApi.
+ */
+class FoiaSubmissionServiceApi implements FoiaSubmissionServiceInterface {
+
+  /**
+   * The Agency Lookup service.
+   *
+   * @var \Drupal\foia_webform\AgencyLookupService
+   */
+  protected $agencyLookUpService;
+
+  /**
+   * An HTTP client to post FOIA requests to an Agency Component API endpoint.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * FoiaSubmissionServiceApi constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   The HTTP Guzzle client.
+   * @param \Drupal\foia_webform\AgencyLookupServiceInterface $agencyLookupService
+   *   The Agency Lookup service.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   */
+  public function __construct(ClientInterface $httpClient, AgencyLookupServiceInterface $agencyLookupService, LoggerInterface $logger) {
+    $this->httpClient = $httpClient;
+    $this->agencyLookUpService = $agencyLookupService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function sendSubmissionToComponent(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    $apiUrl = $agencyComponent->get('field_submission_api');
+    $requestData = NULL;
+
+    if ($apiUrl) {
+      $requestData = $this->assembleRequestData($webformSubmission, $agencyComponent);
+    }
+    else {
+      $this->logger
+        ->error('Unable to submit request via the API. Missing API Submission URL for node %component',
+          [
+            '%component' => $agencyComponent->id(),
+            'link' => $agencyComponent->toLink($this->t('Edit Component'), 'edit-form')->toString(),
+          ]
+        );
+    }
+
+    if ($requestData && $apiUrl) {
+      // @todo Send request here.
+    }
+  }
+
+  /**
+   * Gathers the required fields for the API request.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The FOIA form submission form values.
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   *
+   * @return array
+   *   Return the assemble request data in an array.
+   */
+  public function assembleRequestData(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    // Get the webform submission values.
+    $submissionValues = $this->getSubmissionValues($webformSubmission);
+
+    // If there are files attached, load the files and add the file metadata.
+    if (isset($submissionValues['attachments_supporting_documentation'])) {
```

Comment:
If we ever add an additional attachments field or replace this one, this code will stop working. I think what we probably have to do is search through the submission fields for all of the file fields. Maybe our best bet is to do that within `getAttachmentData` and to invoke `getAttachmentData` from within `getSubmissionValues` so that the code reads cleanly. Checkout our custom `WebformSubmissionResource` inside of `foia_api`, specifically the `getSubmittedFileAttachments` has code that shows how to dynamically find the file fields.